crates/sui-graphql-rpc/src/types/type_filter.rs

```diff
@@ -132,6 +132,31 @@ impl FqNameFilter {
                 .filter(name.eq(n.clone())),
         }
     }
+
+    /// Try to create a filter whose results are the intersection of the results of the input
+    /// filters (`self` and `other`). This may not be possible if the resulting filter is
+    /// inconsistent (e.g. a filter that requires the module member's package to be at two different
+    /// addresses simultaneously), in which case `None` is returned.
+    pub(crate) fn intersect(self, other: Self) -> Option<Self> {
+        use FqNameFilter as F;
+        use ModuleFilter as M;
+
+        match (&self, &other) {
+            (F::ByModule(m), F::ByModule(n)) => m.clone().intersect(n.clone()).map(Self::ByModule),
```

Comment:
an example of this would be coin intersect string, but I don't quite understand what this actually does

Reply:
could we add this as an example

Reply:
There isn't a way to exercise this yet in the GraphQL schema, (there is no way to construct a query that intersects two filters that include fully qualified name filters), so I only updated this for correctness purposes.   Check out the unit tests below for examples of how it would work.

I have a PR coming up that implements the same logic for type filters, and that we may actually be able to exercise in GraphQL, so when I add that, I will be adding E2E tests.